examples/fizzbuzz/main.c

```diff
@@ -0,0 +1,227 @@
+ */
+
+#define UNUSED(x) (void)(x)
+
+#define MIN_RESULT_LEN 9 // len("FizzBuzz") + 1
+
+
+typedef struct fizzbuzz
+{
+    int input;
+    char *result;
+    int result_len;
+    qmlbind_signal_emitter *signalEmitter;
+} fizzbuzz_t;
+
+
+void on_changed(fizzbuzz_t *fb, const qmlbind_value *input)
+{
+    qmlbind_string *inputStr = qmlbind_value_get_string(input);
+    int input_str_len = qmlbind_string_get_length(inputStr);
+    qmlbind_string_release(inputStr);
+    if (input_str_len == 0)
+        return; // Ignore empty textfield
+
+    input_str_len++; // take '\0' into account
+    if (input_str_len > fb->result_len) {
+        fb->result_len =
+            input_str_len > MIN_RESULT_LEN ? input_str_len : MIN_RESULT_LEN;
+        fb->result = realloc(fb->result, sizeof(char) * fb->result_len);
+        if (fb->result == NULL) {
+            printf("Error reallocating memory");
+            qmlbind_application_exit(0);
+        }
+    }
+
+    int i = (int) qmlbind_value_get_number(input);
+    if (i % 15 == 0) {
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "inputWasFizzBuzz",
+                                    0, NULL);
+        strcpy(fb->result, "FizzBuzz");
+    } else if (i % 3 == 0) {
+        strcpy(fb->result, "Fizz");
+    } else if (i % 5 == 0) {
+        strcpy(fb->result, "Buzz");
+    } else {
+        sprintf(fb->result, "%d", i);
+    }
+
+    qmlbind_value *params[1] = {
+        qmlbind_value_new_string_cstr(fb->result)
+    };
+    qmlbind_signal_emitter_emit(fb->signalEmitter, "resultChanged", 1,
+                                (const qmlbind_value * const*) params);
+    qmlbind_value_release(params[0]);
+}
+
+
+qmlbind_client_object *
+fizzbuzz_new_object_cb(qmlbind_client_class *classRef,
+                       qmlbind_signal_emitter *signalEmitter)
+{
+    UNUSED(classRef);
+
+    fizzbuzz_t *fb = malloc(sizeof(fizzbuzz_t));
+    fb->input = 0;
+    fb->result = malloc(sizeof(char) * MIN_RESULT_LEN);
+    fb->result_len = MIN_RESULT_LEN;
+    strcpy(fb->result, "");
+    fb->signalEmitter = signalEmitter;
+
+    return (qmlbind_client_object *) fb;
+}
+
+
+void fizzbuzz_delete_object_cb(qmlbind_client_object *obj)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+    free(fb->result);
+    qmlbind_signal_emitter_release(fb->signalEmitter);
+    free(fb);
+}
+
+
+qmlbind_value *fizzbuzz_call_method_cb(qmlbind_engine *engine,
+                                       qmlbind_client_object *obj,
+                                       const char *name,
+                                       int argc,
+                                       const qmlbind_value *const *argv)
+{
+    UNUSED(engine);
+    UNUSED(argc);
+    UNUSED(argv);
+
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "quit") == 0) {
+        printf("Quitting… Last result was: %s\n", fb->result);
+        qmlbind_application_exit(0); // FIXME: it segfaults :(
```

Comment:
`qmlbind_application_exit()` does not exit program immediately, so probably we need to return `qmlbind_value_undefined()` properly here too. (this fixes the segfault in my environment)